src/py/flwr/cli/new/templates/app/code/flwr_tune/strategy.py.tpl

```diff
@@ -0,0 +1,73 @@
+"""$project_name: A Flower / FlowerTune app."""
+
+from logging import INFO, WARN
+from typing import List, Tuple, Union
+
+from flwr.common import FitIns, FitRes, Parameters, log, parameters_to_ndarrays
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy import FedAvg
+
+
+class FlowerTuneLlm(FedAvg):
+    """Customised FedAvg strategy implementation."""
+    def __init__(self, **kwargs):
+        self.curr_comm_cost = 0.0
+        super().__init__(**kwargs)
+
+    def configure_fit(
+        self, server_round: int, parameters: Parameters, client_manager: ClientManager
+    ):
+        """Configure the next round of training."""
+        return_clients = super().configure_fit(server_round, parameters, client_manager)
+
+        # Test communication costs
+        fit_ins_list = [fit_ins for _, fit_ins in return_clients]
+        self.test_communication_costs(fit_ins_list)
+
+        return return_clients
+
+    def aggregate_fit(
+        self,
+        server_round: int,
+        results: List[Tuple[ClientProxy, FitRes]],
+        failures: List[Union[Tuple[ClientProxy, FitRes], BaseException]],
+    ):
+        """Aggregate fit results using weighted average."""
+        # Test communication costs
+        fit_res_list = [fit_res for _, fit_res in results]
+        self.test_communication_costs(fit_res_list)
+
+        parameters_aggregated, metrics_aggregated = super().aggregate_fit(
+            server_round, results, failures
+        )
+
+        return parameters_aggregated, metrics_aggregated
+
+    def test_communication_costs(self, fit_list: List[Union[FitIns, FitRes]]):
+        """Test communication costs per FL round."""
+
+        def compute_bytes(weights):
+            return sum(ele.nbytes for ele in weights)
+
+        size_bytes_list = [
+            compute_bytes(parameters_to_ndarrays(fit_ele.parameters))
```

Comment:
I think we can make this a more lightweight process by directly counting the number of `bytes` each tensor in the parameters object has. This is an example:

```python
import numpy as np
from flwr.common import ndarrays_to_parameters

arr = np.random.randn(1000,10000)
print(arr.nbytes) # prints 80M
# Let's imagine the above is what FitRes has
p = ndarrays_to_parameters(arr)
p_bytes = sum([len(t) for t in p.tensors])
print(p_bytes) # prints 80.1M
```

`FitRes` is the actual obj being communicated so i wonder if we should count its size directly like above?

Reply:
Ok, thanks for the suggestion. i'm going to use `BytesIO` for bytes count:
```
from io import BytesIO
p_bytes = sum([BytesIO(t).getbuffer().nbytes for t in p.tensors])
```